File: app/actors/AutoScalingDataMonitor.java
package actors;

import java.util.Map;

import com.amazonaws.services.autoscaling.model.AutoScalingGroup;
import com.amazonaws.services.autoscaling.model.LaunchConfiguration;

public interface AutoScalingDataMonitor extends AmazonClientActor
{
    public void monitorAutoScalingData();
    
    public void updateAutoScalingGroupsData();
    
    public void updateLaunchConfigurationsData();
    
    public Map<String, AutoScalingGroup> getAutoScaleData();
    
    public Map<String, LaunchConfiguration> getLaunchConfigurationData();
}

File: app/actors/AutoScalingDataMonitor.scala
package actors

import scala.collection.Map
import com.amazonaws.services.autoscaling.model.AutoScalingGroup
import com.amazonaws.services.autoscaling.model.LaunchConfiguration

trait AutoScalingDataMonitor extends AmazonClientActor {
	
    def monitorAutoScalingData(): Unit;
    
    def updateAutoScalingGroupsData(): Unit;
    
    def updateLaunchConfigurationsData(): Unit;
    
    def getAutoScaleData(): Map[String, AutoScalingGroup];
    
    def getLaunchConfigurationData(): Map[String, LaunchConfiguration];
}